source/MaterialXRuntime/RtLook.cpp

```diff
@@ -170,4 +184,14 @@ RtAttribute RtMaterialAssign::getExclusive() const
     return prim()->getAttribute(EXCLUSIVE)->hnd();
 }
 
+bool RtMaterialAssignConnectableApi::acceptRelationship(const RtRelationship& rel, const RtObject& target) const
```

Comment:
Don't you also need a material?

Reply:
I have changed the material port to be an input of "material" type instead of a relationship. So connectability for that port is handled by the acceptConnection method in the base class.